pkg/performanceprofile/controller/performanceprofile_controller.go

```diff
@@ -449,16 +450,61 @@ func (r *PerformanceProfileReconciler) Reconcile(ctx context.Context, req ctrl.R
 	})
 	err = r.Client.Get(context.Background(), key, nodeCfg)
 	if err != nil {
-		klog.Errorf("failed to get config node object; name=%q err=%v", nodeCfg.GetName(), err)
-		nodeCfg.Name = nodeCfgName
-		nodeCfg.Spec.CgroupMode = apiconfigv1.CgroupModeV1
-		//nolint:errcheck
-		r.Client.Update(ctx, nodeCfg)
+		return reconcile.Result{}, err
 	}
+
 	if nodeCfg.Spec.CgroupMode != apiconfigv1.CgroupModeV1 {
+		// gather the MCP generation
+		profileMCP, err := r.getMachineConfigPoolByProfile(ctx, instance)
+		if err != nil {
+			return reconcile.Result{}, err
+		}
+		startGeneration := profileMCP.Generation
+
 		nodeCfg.Spec.CgroupMode = apiconfigv1.CgroupModeV1
-		//nolint:errcheck
-		r.Client.Update(ctx, nodeCfg)
+		if err := r.Client.Update(ctx, nodeCfg); err != nil {
+			return reconcile.Result{}, err
+		}
+
+		// wait for MCP generation to change
+		err = wait.PollUntilContextCancel(context.Background(), time.Duration(7*time.Second), true, func(ctx context.Context) (done bool, err error) {
+			profileMCP, err := r.getMachineConfigPoolByProfile(ctx, instance)
+			if err != nil {
+				return true, err
+			}
+			// check to see if machine rollout has happened
+			if startGeneration != profileMCP.Generation {
+				return true, nil
+			}
+			// check to see if all machines are updating
+			if profileMCP.Status.DegradedMachineCount > 0 {
+				return true, fmt.Errorf("machines have become degraded")
+			}
+			return false, nil
+		})
+		if err != nil {
+			return reconcile.Result{}, err
+		}
+
+		// wait for Cgroupv1 roll out
+		err = wait.PollUntilContextCancel(context.Background(), time.Duration(7*time.Second), true, func(ctx context.Context) (done bool, err error) {
+			profileMCP, err := r.getMachineConfigPoolByProfile(ctx, instance)
+			if err != nil {
+				return true, err
+			}
+			// check to see if all machines are updated
+			if profileMCP.Status.MachineCount == profileMCP.Status.ReadyMachineCount {
+				return true, nil
+			}
+			// check to see if all machines are updated
+			if profileMCP.Status.DegradedMachineCount > 0 {
+				return true, fmt.Errorf("machines have become degraded")
+			}
+			return false, nil
+		})
+		if err != nil {
+			return reconcile.Result{}, err
+		}
```

Comment:
ok, reviewing again this is meant to wait at most 7+7 seconds, right?
if so, as it looks to me, it could be OK for the short term because a better fix would require more invasive changes, which we should plan and implement later (and backport).
The only missing thing is: how was 7 seconds determined and how it (if it all) interacts with the controller reconcile loop? e.g. there is any relationship between the two or not?

@MarSik @yanirq @jmencak any concern about accepting this wait as short-term solution, provided a wait-less solution is implemented as followup?

Reply:
a later commit removes the second wait, even better!

Reply:
Hmm, we are basically only waiting until MCP notices the cgroups change, right? That might be an acceptable evil here.

What happens when the 7+7 seconds pass? We return a reconcile error and try again later?

Reply:
The wait loop retries every 7 seconds to look for a MCP generation change.

Reply:
@ffromani I was thinking about the status solution. Can't we just return when the MCP is not ready here and rely on the watch that we already have (we watch MCP, right?) to rerun the Reconcile method once something changes?